src/ucp/core/ucp_ep.c

```diff
@@ -3133,7 +3153,8 @@ static void ucp_ep_req_purge_send(ucp_request_t *req, ucs_status_t status)
 
     if ((ucp_ep_config(req->send.ep)->key.err_mode !=
          UCP_ERR_HANDLING_MODE_NONE) &&
-        (req->flags & UCP_REQUEST_FLAG_RKEY_INUSE)) {
+        (req->flags & UCP_REQUEST_FLAG_RKEY_INUSE) &&
+        !(req->flags & UCP_REQUEST_FLAG_USER_MEMH)) {
```

Comment:
how will invalidate flow work with user memh?

Reply:
user has to sync out-of-band before doing ucp_mem_unmap of a memh that was provided to some send/recv operation
(similar to rma/amo requirements for rkey)

Reply:
does it mean error handling will not be fully supported with user defined memh?

Reply:
yes, in the sense that user cannot release the memh that was passed to ucp request only because the local ep is destroyed. should also make sure the remote ep was destroyed (by additional protocol, or just never destroy the memh)
in order to really support it probably need to create an internal indirect key on top of user memh